spawn.lua

```diff
@@ -1,66 +1,45 @@
+function SendToWorldSpawn( Player )
+
+	local World = Player:GetWorld()
+	Player:TeleportToCoords(World:GetSpawnX(), World:GetSpawnY(), World:GetSpawnZ())
+
+end
+
 function HandleSpawnCommand(Split, Player)
 	
-	local WorldIni = cIniFile()
-	WorldIni:ReadFile(Player:GetWorld():GetIniFileName())
-	
-	local SpawnX = WorldIni:GetValue("SpawnPosition", "X")
-	local SpawnY = WorldIni:GetValue("SpawnPosition", "Y")
-	local SpawnZ = WorldIni:GetValue("SpawnPosition", "Z")
-	local flag = 0
-	
-	if (#Split == 2 and Split[2] ~= Player:GetName()) then
+	if ( (#Split == 2) and (Split[2] ~= Player:GetName()) ) then
 		if Player:HasPermission("core.spawn.others") then
+			local PlayerFound = false
 			local FoundPlayerCallback = function(OtherPlayer)
 				if (OtherPlayer:GetName() == Split[2]) then
-					World = OtherPlayer:GetWorld()
-					local OnAllChunksAvaliable = function()
-						OtherPlayer:TeleportToCoords(SpawnX, SpawnY, SpawnZ)
-						SendMessageSuccess( Player, "Returned " .. OtherPlayer:GetName() .. " to world spawn" )
-						flag=1
-					end
-					World:ChunkStay({{SpawnX/16, SpawnZ/16}}, OnChunkAvailable, OnAllChunksAvaliable)		
+					SendToWorldSpawn( OtherPlayer )
+					SendMessageSuccess( Player, "Returned " .. OtherPlayer:GetName() .. " to " .. OtherPlayer:GetWorld():GetName() .. "'s spawn" )
+					PlayerFound = true
 				end
 			end
 			cRoot:Get():FindAndDoWithPlayer(Split[2], FoundPlayerCallback)
 			
-			if flag == 0 then
+			if ( PlayerFound == false ) then
```

Comment:
Use `not(PlayerFound)´, then you won't need initialization to ´false´.

Also consider renaming to ´IsPlayerFound´.